### PR TITLE
feat(faceted-search): replace faceted mode toggle by design system switch component

### PR DESCRIPTION
**Pull request by AdelineMrtn**
**What is the problem this PR is trying to solve?** 
- Added a new switch component in order to improve the UX of the current basic/advanced faceted. 
- The goal is to have a better readability and understanding thanks to this new switch component. 

**What is the chosen solution to this problem?**
- Import "@talend/design-system" in order to use the new toggle form component from design system library.
- Adapt the props of this new <Form.Switch/> components.

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**

### CODE REVIEW

File: packages/faceted-search/.storybook/preview.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-
+import {ThemeProvider} from '@talend/design-system';
```
[the remark this answers is not in the document]
Comment (Kporal) [reply packages/faceted-search/.storybook/preview.js:2]:
```suggestion
import { ThemeProvider } from '@talend/design-system';
```

File: packages/faceted-search/src/components/FacetedToolbar/FacetedToolbar.component.js [hunk 2]
```diff
@@ -1,33 +1,22 @@
...
-	);
-};
+const SwitchFacetedMode = ({ facetedMode, onChange, t }) => (
+	<form>
+		<Form.Switch
+			label={t('FACETED_SEARCH_QUERY', 'Query')}
+			onChange={() => onChange(facetedMode === FACETED_MODE.BASIC ? FACETED_MODE.ADVANCED : FACETED_MODE.BASIC)}
+			dataFeature={facetedMode === FACETED_MODE.BASIC ? USAGE_TRACKING_TAGS.ADVANCED : USAGE_TRACKING_TAGS.BASIC}
+		/>
+	</form>
```
Comment (frassinier) [inline packages/faceted-search/src/components/FacetedToolbar/FacetedToolbar.component.js:18]: You can use the Form component from the DS too
```suggestion
	<Form>
		<Form.Switch
			label={t('FACETED_SEARCH_QUERY', 'Query')}
			onChange={() => onChange(facetedMode === FACETED_MODE.BASIC ? FACETED_MODE.ADVANCED : FACETED_MODE.BASIC)}
			dataFeature={facetedMode === FACETED_MODE.BASIC ? USAGE_TRACKING_TAGS.ADVANCED : USAGE_TRACKING_TAGS.BASIC}
		/>
	</Form>
```

File: packages/faceted-search/src/components/FacetedToolbar/FacetedToolbar.component.js [hunk 3]
```diff
@@ -1,33 +1,22 @@
...
-			<Toggle.Label values={values} value={facetedMode} onChange={onChange} />
-		</form>
-	);
-};
+const SwitchFacetedMode = ({ facetedMode, onChange, t }) => (
+	<Form>
+		<Form.Switch
+			label={t('FACETED_SEARCH_QUERY', 'Query')}
+			onChange={() => onChange(facetedMode === FACETED_MODE.BASIC ? FACETED_MODE.ADVANCED : FACETED_MODE.BASIC)}
+			dataFeature={facetedMode === FACETED_MODE.BASIC ? USAGE_TRACKING_TAGS.ADVANCED : USAGE_TRACKING_TAGS.BASIC}
```
Comment (ncomont) [inline packages/faceted-search/src/components/FacetedToolbar/FacetedToolbar.component.js:16]: shouldn't it be the opposite ?

**Review by ncomont — CHANGES_REQUESTED** [review]

File: packages/faceted-search/src/components/FacetedToolbar/FacetedToolbar.component.js (hunk 3 above)
Comment (AdelineMrtn) [reply packages/faceted-search/src/components/FacetedToolbar/FacetedToolbar.component.js:16]: what do you mean by that please ?

**Review by Nigui — APPROVED** [review]

**Pull request merged**